sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -76,7 +76,7 @@ statement
     | ctes? dmlStatementNoWith                                         #dmlStatement
     | USE identifierReference                                          #use
     | USE namespace identifierReference                                #useNamespace
-    | SET CATALOG (identifier | stringLit)                             #setCatalog
+    | SET CATALOG (errorCapturingIdentifier | stringLit)                  #setCatalog
```

Comment:
Do we consider extending:
```
IDENTIFIER
    : (UNICODE_LETTER | DIGIT | '_')+
    | UNICODE_LETTER+ '://' (UNICODE_LETTER | DIGIT | '_' | '/' | '-' | '.' | '?' | '=' | '&' | '#' | '%')+
```
to 
```
IDENTIFIER
    : (UNICODE_LETTER | DIGIT | '_' | '-')+
    | UNICODE_LETTER+ '://' (UNICODE_LETTER | DIGIT | '_' | '/' | '-' | '.' | '?' | '=' | '&' | '#' | '%')+
```
so that we can remove all errorCapturingIdentifier for simplicity.

Reply:
We cannot. that would break SELECT c1 - c2 FROM VALUES(1, 2) AS T(c1, c2);

An interesting thought would be to flip the script and rename errorCapturingIdentifier to identifier.
And find another name for identifier like identifierIsExpression.
But that adds churn.